java/src/jmri/jmrix/AbstractMRTrafficController.java

```diff
@@ -1094,6 +1098,38 @@ public void run() {
     }  // end XmtNotifier
 
     /**
+     * Terminate the receive and transmit threads.
+     *<p>
+     * This is intended to be used only by testing subclasses.
+     */
+    public void terminateThreads() {
+        threadStopRequest = true;
+        if (xmtThread != null) {
+            xmtThread.stop();
```

Comment:
I think these really should be interupt calls instead of stops as well.  Thread.stop() is deprecated because it is inherently unsafe ( per oracle).

Even for a test, we don't want to introduce new deprecated calls here.

The threadStopRequest flag will cause the loop in the run method to terminate, which will cause the thread to finish ( which I'd what PR #3550 does as well, though without the first thread interrupt. ).